addons/mail/controllers/main.py

```diff
@@ -112,7 +112,12 @@ def receive(self, req):
     def read_followers(self, follower_ids):
         result = []
         is_editable = request.env.user.has_group('base.group_no_one')
-        for follower in request.env['mail.followers'].browse(follower_ids):
+        follower_recs = request.env['mail.followers'].sudo().browse(follower_ids)
+        res_ids = follower_recs.mapped('res_id')
+        if follower_recs:
+            res_model = follower_recs[0].res_model
+            request.env[res_model].browse(res_ids).check_access_rule("write")
```

Comment:
Last comment: 
This bugfix causes errors when the user accesses a read-only record, for example a Leave that is already validated. 
I think read_followers() should only check for 'read' access, not 'write' access, as it does not change the followers.